eth/api.go

```diff
@@ -1506,17 +1508,58 @@ func NewPrivateDebugAPI(eth *Ethereum) *PrivateDebugAPI {
 	return &PrivateDebugAPI{eth: eth}
 }
 
-// ProcessBlock reprocesses an already owned block.
-func (api *PrivateDebugAPI) ProcessBlock(number uint64) (bool, error) {
+// Block processes the given block's RLP but does not import the block in to
+// the chain.
+func (api *PrivateDebugAPI) Block(blockRlp []byte, logger VmLoggerOptions) (bool, error) {
+	var block types.Block
+	err := rlp.Decode(bytes.NewReader(blockRlp), &block)
+	if err != nil {
+		return false, fmt.Errorf("could not decode block: %v", err)
+	}
+
+	return api.debugBlock(&block, logger)
+}
+
+// BlockFromFile loads the block's RLP from the given file name and attempts to
+// process it but does not import the block in to the chain.
+func (api *PrivateDebugAPI) BlockFromFile(fn string, logger VmLoggerOptions) (bool, error) {
+	file, err := os.Open(fn)
+	if err != nil {
+		return false, fmt.Errorf("could not load file: %v", err)
+	}
+
+	blockRlp, err := ioutil.ReadAll(file)
+	if err != nil {
+		return false, fmt.Errorf("could not read file: %v", err)
+	}
```

Comment:
I always forget about that function. Reminds of a discussion @fjl and I had a long time ago.